ludwig/models/abstractmodel.py

```diff
@@ -0,0 +1,242 @@
+import logging
+from abc import ABCMeta, abstractmethod
+from collections import OrderedDict
+from typing import Any, Dict, List, Optional, Tuple, Union
+
+import numpy as np
+import torch
+
+from ludwig.combiners.combiners import Combiner
+from ludwig.constants import COMBINED, LOSS, NAME, TIED, TYPE
+from ludwig.features.base_feature import InputFeature, OutputFeature
+from ludwig.features.feature_registries import input_type_registry, output_type_registry
+from ludwig.utils.algorithms_utils import topological_sort_feature_dependencies
+from ludwig.utils.metric_utils import get_scalar_from_ludwig_metric
+from ludwig.utils.misc_utils import get_from_registry
+from ludwig.utils.torch_utils import LudwigModule, reg_loss
+
+logger = logging.getLogger(__name__)
+
+
+class AbstractModel(LudwigModule, metaclass=ABCMeta):
+    def __init__(self, random_seed=None):
+        self._random_seed = random_seed
+
+        # TODO: with change to misc_utils.set_random_seed() this may be redundant
+        #       seems to be required for test_api.py::test_api_training_determinism
+        if random_seed is not None:
+            torch.random.manual_seed(random_seed)
```

Comment:
Is this relevant for GBM models as well?

Reply:
GBM loads tree weights into PyTorch using Hummingbird. I'm not sure if after that, there are still random operations